lab-jonathan/test/test-chat.js

```diff
@@ -0,0 +1,54 @@
+let server = require('server.js');
+let net = require('net');
+var assert = require('chai').assert;
+
+let nickname = require('server.js').nickname;
+//test server starts
+describe('test the server connection', function(){
+  before(function() {
+    server.listen(3000);
+  });
+  describe('server behavior', function(){
+    before(function(done){
+      let client;
+      client = net.createConnection({port: 3000});
+      done();
+    })
+    it('should start', function(done){
+      console.log('it started!' + client);
```

Comment:
I'm guessing that `client` here comes back undefined? Up on line 13, you define and assign client inside the scope of the before block. Move line 13 up to line 12, and you should then have access to the client object.